hack/prowimagebuilder/main.go

```diff
@@ -186,17 +186,24 @@ func allTags(arch string) ([]string, error) {
 	return allTags, nil
 }
 
+var datePrefix string
+
 // gitTag returns YYYYMMDD-<GIT_TAG>
+// In order to ensure a consistent date value across the runtime of this process
+// when run near midnight UTC, we cache the value of date.
+// We don't cache the git SHA since a change in that would be meaningful.
 func gitTag() (string, error) {
-	prefix, err := runCmd(nil, "date", "+v%Y%m%d")
-	if err != nil {
-		return "", err
+	var err error
+	if datePrefix == "" {
+		if datePrefix, err = runCmd(nil, "date", "+v%Y%m%d"); err != nil {
```

Comment:
Why not just use Go's time utility to generate a date (is there really some advantage to shelling out to `date`)?

Reply:
That sounds better.
I'm just trying to make the fastest change to fix this though and don't want to think about date formatting.